SDK/AppCenterCrashes/Microsoft.AppCenter.Crashes.Windows.Shared/Crashes.cs

```diff
@@ -200,31 +203,73 @@ private Task ProcessPendingErrorsAsync()
+                        AppCenterLog.Debug(LogTag, "The flag for user confirmation is set to AlwaysSend, will send logs.");
+                        await HandleUserConfirmationAsync(UserConfirmation.Send);
+                        return;
+                    }
+
+                    if (ShouldAwaitUserConfirmation?.Invoke() ?? false)
+                    {
+                        AppCenterLog.Debug(LogTag, "ShouldAwaitUserConfirmation returned true, wait sending logs.");
+                    }
+                    else
+                    {
+                        AppCenterLog.Debug(LogTag, "ShouldAwaitUserConfirmation returned false, will send logs.");
+                        await HandleUserConfirmationAsync(UserConfirmation.Send);
+                    }
+                }
+            });
         }
 
-        private Task HandleUserConfirmationAsync()
+        private Task HandleUserConfirmationAsync(UserConfirmation userConfirmation)
         {
-            // Send every pending log.
             var keys = _unprocessedManagedErrorLogs.Keys.ToList();
             var tasks = new List<Task>();
-            foreach (var key in keys)
+
+            if (userConfirmation == UserConfirmation.DontSend)
             {
-                var log = _unprocessedManagedErrorLogs[key];
-                tasks.Add(Channel.EnqueueAsync(log));
-                _unprocessedManagedErrorLogs.Remove(key);
-                ErrorLogHelper.RemoveStoredErrorLogFile(key);
-                var errorReport = new ErrorReport(log, null);
-
-                // This must never called while a lock is held.
-                var attachments = GetErrorAttachments?.Invoke(errorReport);
-                if (attachments == null)
+                foreach (var key in keys)
                 {
-                    AppCenterLog.Debug(LogTag, $"Crashes.GetErrorAttachments returned null; no additional information will be attached to log: {log.Id}.");
+                    _unprocessedManagedErrorLogs.Remove(key);
+                    ErrorLogHelper.RemoveStoredErrorLogFile(key);
+                    // TODO: Remove exception files
                 }
-                else
+            }
+            else
+            {
+                if (userConfirmation == UserConfirmation.AlwaysSend)
+                {
+                    ApplicationSettings.SetValue(PrefKeyAlwaysSend, true);
+                }
+
+                // Send every pending log.
+                foreach (var key in keys)
                 {
-                    tasks.Add(SendErrorAttachmentsAsync(log.Id, attachments));
+                    var log = _unprocessedManagedErrorLogs[key];
+                    tasks.Add(Channel.EnqueueAsync(log));
+                    _unprocessedManagedErrorLogs.Remove(key);
+                    ErrorLogHelper.RemoveStoredErrorLogFile(key);
+                    var errorReport = new ErrorReport(log, null);
+
+                    // This must never called while a lock is held.
```

Comment:
Totally unrelated but I'm just noticing a typo (my bad!):
```suggestion
                    // This must never be called while a lock is held.
```